.ci/integration-tests

```diff
@@ -288,6 +356,19 @@ function tc_machine_deployment() {
     printf "\nCompleted TestCase\n"
 }
 
+function tc_check_orphan_resources() {
+    
+    printf "\nStarting TestCase: %s" "${FUNCNAME[0]}"
+     
+    # Check if there are any leftover resources.
+    if ! hf_orphan_resources "$provider"; then
+        printf "\n\t Failed: There are orphan resources in the cloud. Exiting the test."
```

Comment:
Also, should we delete these orphan resources if/when we find them?

Reply:
Sure, we can do that, I'll adapt it.

Reply:
done - https://github.com/gardener/machine-controller-manager/pull/550/commits/5c9b6d9bfbe1a621e7c459774efa8f830f280adf